dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+                    }
+                    catInodes = catbuilder.toString();
+
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("FilteredCategories=$categories.filterCategoriesByUserPermissions([")
+                        .append(catInodes)
+                        .append("] ))");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("Categories=$categories.fetchCategoriesInodes($")
+                        .append(field.variable())
+                        .append("FilteredCategories))");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesNames=$categories.fetchCategoriesNames($")
+                        .append(field.variable())
+                        .append("FilteredCategories))");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("=$")
+                        .append(field.variable())
+                        .append("Categories)");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesKeys=$categories.fetchCategoriesKeys($")
+                        .append(field.variable())
+                        .append("FilteredCategories))");
+                } else {
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("FilteredCategories=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("Categories=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesNames=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesKeys=$contents.getEmptyList())");
+                }
+            } else if (field instanceof TagField) {
+                content.setTags();
+                String value = content.getStringProperty(field.variable());
+                sb.append("#set($")
+                    .append(field.variable())
+                    .append("=\"")
+                    .append(UtilMethods.espaceForVelocity(value)
+                        .trim())
+                    .append("\")");
+            }
+        }
+
+
+        // get the contentlet categories to make a list
+        String categories = "";
+        Set<Category> categoryList = new HashSet<Category>(categoryAPI.getParents(content, systemUser, false));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)